borsh/src/lib.rs

```diff
@@ -100,8 +102,21 @@ pub mod error;
 #[cfg(all(feature = "std", feature = "hashbrown"))]
 compile_error!("feature \"std\" and feature \"hashbrown\" don't make sense at the same time");
 
+#[cfg(feature = "std")]
+use std::io as io_impl;
 #[cfg(not(feature = "std"))]
-pub mod nostd_io;
+mod nostd_io;
+#[cfg(not(feature = "std"))]
+use nostd_io as io_impl;
+
+/// Subset of `std::io` which is used as part of borsh public API.
+///
+/// When crate is built with `std` feature disabled (it’s enabled by default),
+/// the exported types are custom borsh types which try to mimic behaviour of
+/// corresponding standard types usually offering subset of features.
+pub mod io {
+    pub use super::io_impl::{Error, ErrorKind, Read, Result, Write};
```

Comment:
rust allows to do this without much confusion ))
![tmp](https://github.com/near/borsh-rs/assets/26653921/87a7a7a3-0acf-4c43-9295-5d04e92293af)